src/core/handlers/GraphQLHandler.ts

```diff
@@ -141,11 +144,17 @@ export class GraphQLHandler extends RequestHandler<
     const match = matchRequestUrl(new URL(args.request.url), this.endpoint)
     if (!match.matches) return { match }
 
-    const parsedResult = await parseGraphQLRequest(args.request).catch(
-      (error) => {
-        console.error(error)
-        return undefined
-      },
+    if (!GraphQLHandler.#parseGraphQLRequestCache.has(args.request)) {
```

Comment:
I think caching the request parsing result is something all request handlers would want to benefit from. Does it make sense to move this logic to the `RequestHandler` base class so the child classes don't have to manage this manually?

Reply:
I think we should instead pass the request clone to the handlers from higher up the tree, but I wasn't sure if that's a reasonable option, since it changes the args to `handler.run()` slightly?  This was mostly to show how many fewer clones can be made with a few changes

Reply:
Actually, a few things.  I was thinking of the other cache point at first

This graphql parsing has 2 callsites. 1 per handler and 1 again when no call is matched.

Ideally, we'd only parse once per request and not in both cases, where it might be more reasonable to keep the version of `parseGrapqhlRequestHandler` internal to msw as a version that caches results (we'd probably expose the version that doesn't cache - but could expose a caching version). 


I'd like to have parse be cached by default, but haven't quite figured out the right api to make that ergononmic as a general feature for `RequestHandler` because there's 2 separate types of request parsing that happens - one that uses handler specific info (endpoint) and one that doesn't (request body).

In the intermediate, handling something along these lines is probably good enough, and keeping them private should allow us a bit of internal freedom to optimize/improve over time before exposing something more generally?

Reply:
We could break parsing into `parse endpoint` and `parse body` type calls, combining their outputs into parse, but this expands the surface area in a way we might not want to?

only body parsing would likely be cached in that split

Reply:
Agreed. Let's keep the changes surface to a minimum. This is already a grand improvement as it is.